modules/aws/vpc/main.tf

```diff
@@ -13,23 +13,23 @@ terraform {
   }
 }
 
-resource "aws_flow_log" "flow_log" {
-  iam_role_arn    = aws_iam_role.iam_role[0].arn
-  log_destination = aws_cloudwatch_log_group.log_group[0].arn
+resource "aws_flow_log" "this" {
+  iam_role_arn    = aws_iam_role.this[0].arn
+  log_destination = aws_cloudwatch_log_group.this[0].arn
   traffic_type    = var.vpc_flow_logs_traffic_type
-  vpc_id          = aws_vpc.vpc.id
+  vpc_id          = aws_vpc.this.id
   count           = var.enable_vpc_flow_logs ? 1 : 0
```

Comment:
[count should be the first item](https://www.terraform-best-practices.com/naming#usage-of-count-for_each)